tests/features/scheduling/test_scheduling.py

```diff
@@ -0,0 +1,160 @@
+
+        time = pendulum.now().on(2018, 5, 21).at(3, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == True
+
+        self.task.run_every = "2 hours"
+        time = pendulum.now().on(2018, 5, 21).at(2, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == True
+
+        self.task.run_every = "2 hours"
+        time = pendulum.now().on(2018, 5, 21).at(3, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+        time = pendulum.now().on(2018, 5, 21).at(4, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == True
+
+    def test_scheduler_should_run_every_days(self):
+        self.task.run_every = "2 days"
+        time = pendulum.now().on(2018, 5, 21).at(0, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+        time = pendulum.now().on(2018, 5, 23).at(0, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+        self.task.run_at = "5:30"
+        time = pendulum.now().on(2018, 5, 22).at(5, 30, 0)
+        self.task._date = time
+        assert self.task.should_run(time) == True
+
+        self.task.run_at = "5:35"
+        time = pendulum.now().on(2018, 5, 22).at(5, 30, 0)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+    def test_scheduler_should_run_every_months(self):
+        self.task.run_every = "2 months"
+        time = pendulum.now().on(2018, 1, 1).at(0, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+        time = pendulum.now().on(2018, 2, 1).at(0, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == True
+
+        time = pendulum.now().on(2018, 2, 1).at(10, 0, 1)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+        self.task.run_at = "5:30"
+        time = pendulum.now().on(2018, 2, 1).at(5, 30, 0)
+        self.task._date = time
+        assert self.task.should_run(time) == False
+
+    def test_twice_daily_at_correct_time(self):
+        time = pendulum.now().on(2018, 1, 1).at(1, 20, 5)
+        self.task.run_every = ""
+        self.task.twice_daily = (1, 13)
+        self.task._date = time
+
+        assert self.task.should_run()
+
+        time = pendulum.now().on(2018, 1, 1).at(13, 20, 5)
+        self.task._date = time
+        assert self.task.should_run()
+
+    def test_twice_daily_at_incorrect_time(self):
+        time = pendulum.now().on(2018, 1, 1).at(12, 20, 5)
+        self.task.run_every = ""
+        self.task.twice_daily = (1, 13)
+        self.task._date = time
+
+        assert self.task.should_run() is False
+
+    def test_run_at(self):
+        self.task.run_every = ""
+        self.task.run_at = None
+        self.task.run_at = "13:00"
+
+        time = pendulum.now().on(2018, 1, 1).at(13, 0, 5)
```

Comment:
I am not sure of what you're trying to do here. Are some asserts missing ?

Reply:
just setting the time to 1pm